### DISCUSSION TITLE
Practice mode idea (+about NC)

### DISCUSSION THREAD

**Discussion by Givikap120** [ppy/osu#20323] [Ideas]
What about adding mod "Cut", that crops beatmap from point A to point B, chosen in Mod Customisation?

Add Local mod presets and a possibility to make "dummy difficulty" based on mod preset (so you can use preset HDDT + Cut for particular map difficulty by choosing from the "train dummy difficulties" menu.

Also, there were similar topics, but. Separate "speed adjust" and "pitch adjust" in DT/HT. I use NC because it sounds much cleaner and more natural, but adjusting speed in Mod Customisation ruins that.

**Comment by ghost** [comment]
(I have a demo of the pitch adjust mod coded up, but the code quality stinks)

I'm not sure how you expect this "cut" mod to work. Where would you select point A and point B? In mod customization? Would you input both timestamps manually?

**Reply by Givikap120** [reply]
Yes, in Mod Customization:
Start point - 32.4
End Point - 123.7
In seconds

**Reply by ghost** [reply]
Wouldn't it be easier to just go into the editor and make a practice diff? You'd probably have to go to the editor to get the timestamps of each object anyways, so why not make the diff while youre there?

**Reply by Givikap120** [reply]
1) Because practice diff will stand with other diffs, I don't like this: looks bad and impairs navigation between maps.
2) Because it's unsubmitted: you can't play it in multi, you can't share replays, and it doesn't count in profile statistics.
3) You can add something like a strain graph preview, so you doesn't forced to go to the editor.
